src/app/storage/AttributeStorage.h

```diff
@@ -0,0 +1,161 @@
+/**
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app/ConcreteAttributePath.h>
+#include <app/storage/PascalString.h>
+#include <lib/core/CHIPError.h>
+#include <lib/support/Span.h>
+
+namespace chip {
+namespace app {
+namespace Storage {
+
+/// Defines an interface for persisting attribute values.
+///
+/// Provides read/write of values (char/byte arrays or numeric values) based
+/// on a endpoint/cluster/attribute path.
+///
+/// To take into account storage compatibility, the class uses the concepts of:
+///
+///  - `Value` represents something that is writable to persistent storage
+///  - `Buffer` represents something that can be read from persistent storage
+///
+class AttributeStorage
+{
+private:
+    static ByteSpan asByteSpan(CharSpan span) { return { reinterpret_cast<const uint8_t *>(span.data()), span.size() }; }
+    static ByteSpan asByteSpan(ByteSpan span) { return span; }
+
+public:
+    /// Represents an attribute value that can be written to storage.
+    ///
+    /// In particular it has information about the data content since
+    /// historical ember-specific writes for strings encode data size as
```

Comment:
In this context, it's really unclear what "ember" is even talking about.  I think it's better to just say this is for backwards compatibility with how clusters used to persist their data and leave it at that?